sled-agent/src/rack_setup/service.rs

```diff
@@ -1121,126 +1149,91 @@ impl ServiceInner {
+                boundary_switch_addrs.insert(
+                    *switch_mgmt_addrs.get(&uplink_config.switch).unwrap(),
+                );
 
-            // TODO handle breakouts
-            // https://github.com/oxidecomputer/omicron/issues/3062
-            let link_id = LinkId(0);
-            let addr = IpAddr::V4(rack_network_config.uplink_ip);
-
-            let link_settings = LinkSettings {
-                // TODO Allow user to configure link properties
-                // https://github.com/oxidecomputer/omicron/issues/3061
-                params: LinkCreate {
-                    autoneg: false,
-                    kr: false,
-                    fec: convert_fec(&rack_network_config.uplink_port_fec),
-                    speed: convert_speed(
-                        &rack_network_config.uplink_port_speed,
-                    ),
-                },
-                addrs: vec![addr],
-            };
+                let (ipv6_entry, dpd_port_settings, port_id) =
+                    self.build_uplink_config(uplink_config)?;
 
-            dpd_port_settings.links.insert(link_id.to_string(), link_settings);
+                self.wait_for_dendrite(dpd).await;
 
-            let port_id: PortId = rack_network_config.uplink_port.parse()
-                .map_err(|e| SetupServiceError::BadConfig(
-                        format!("could not use value provided to rack_network_config.uplink_port as PortID: {e}")
-                ))?;
+                info!(self.log, "Configuring boundary services loopback address on switch"; "config" => format!("{ipv6_entry:#?}"));
+                dpd.loopback_ipv6_create(&ipv6_entry).await.map_err(|e| {
+                    SetupServiceError::Dendrite(format!(
+                        "unable to create inital switch loopback address: {e}"
+                    ))
+                })?;
 
-            let nexthop = Some(rack_network_config.gateway_ip);
+                info!(self.log, "Configuring default uplink on switch"; "config" => format!("{dpd_port_settings:#?}"));
+                dpd.port_settings_apply(&port_id, &dpd_port_settings)
+                    .await
+                    .map_err(|e| {
+                        SetupServiceError::Dendrite(format!(
+                    "unable to apply initial uplink port configuration: {e}"
+                ))
+                    })?;
 
-            dpd_port_settings.v4_routes.insert(
-                Ipv4Cidr { prefix: "0.0.0.0".parse().unwrap(), prefix_len: 0 }
-                    .to_string(),
-                RouteSettingsV4 {
-                    link_id: link_id.0,
-                    vid: rack_network_config.uplink_vid,
-                    nexthop,
-                },
-            );
+                info!(
+                    self.log,
+                    "advertising boundary services loopback address"
+                );
+                let zone_addr =
+                    switch_mgmt_addrs.get(&uplink_config.switch).unwrap();
```

Comment:
move this up a lil bit so we can use it on line 1185 as well?

Reply:
Done